RibbonPhotos/Presentations/SplashViewController.swift

```diff
@@ -49,15 +51,36 @@ final class SplashViewController: UIViewController {
     private func showUserScenario() {
         if oauth2TokenStorage.bearerToken != nil {
             switchToTabBarController()
+            fetchProfile(token: oauth2TokenStorage.bearerToken ?? "")
```

Comment:
по тз, надо запросить данные профиля, но вот в чем беда, ты перед этим свитчуешь на главный экран. Это действие выгружает сплеш скрин из памяти, и если имя профиля еще как-то успевает синхронизироваться, то к моменту когда нужно дозапросить картинку, self уже не существует и ты теряешь запрос (строки с 64 по 70) 

Во вторую сессию пользователь остается без картинок. 

Поэтому я бы здесь написала немного иначе, начала всегда фетчить профиль, а свитч вызывала бы только когда профиль загружен и ушел запрос на аватарку 

```suggestion
        if let token = oauth2TokenStorage.bearerToken {
            fetchProfile(token: token)
```